src/manager/msWindowManager.js

```diff
@@ -9,6 +9,11 @@ const { MsWindow } = Me.imports.src.layout.msWorkspace.msWindow;
 const { MsDndManager } = Me.imports.src.manager.msDndManager;
 const { getSettings } = Me.imports.src.utils.settings;
 
+// Wayland takes much longer to display new windows, so the way is increased
+const isWayland = GLib.getenv('XDG_SESSION_TYPE').toLowerCase() === 'wayland';
+const MAX_TIME_FIND = 2000 + ((isWayland ? 2 : 0) * 1000);
+const MAX_TIME_WAIT = 5000 + ((isWayland ? 2 : 0) * 1000);
```

Comment:
I'm not sure this should be related to wayland. I guess is more a low end / high end computer. If you were using Xorg all day long you should face the same issue

Reply:
I don't have problems running MS on XOrg, the errors in log and problems opening app only happen in Wayland (new windows always take longer to show up on Wayland).